caom2pipe/reader_composable.py

```diff
@@ -212,6 +214,43 @@ def set_headers(self, storage_name):
         self._logger.debug('End set_headers')
 
 
+class DelayedClientReader(StorageClientReader):
+    """Use case: TaskType.STORE, with use_local_files set to False.
+
+    The object is to be able to delay the retrieval of the FileInfo and header information until the file is retrieved
```

Comment:
Refactored so that the functions `_retrieve_headers` and `_retrieve_file_info` now are the complete functionality in the `for` loop of the `set_file_info` and `set_headers` calls.